### ISSUE TITLE
WebUI reverts to SSL off and UPnP enabled on container restart

### ISSUE THREAD

**Issue by pv2b** [linuxserver/docker-qbittorrent#144]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior
I want to be able to run qBittorrent using SSL. This is required e.g. for registering qBittorrent as a magnet link handler.

For this purpose, I have provided certificates in a volume accessible from the container. I am then trying to use the WebUI to change to HTTPS.

I also want to disable UPnP / NAT-PMP for the WebUI since I do not want the qbittorrent webui exposed through my router on my public IP.

Therefore I want to change the configuration using the qBittorrent WebUI.

## Current Behavior

This will work until qbittorrent is restarted. At that point, the SSL setting is reverted to non-SSL. Also, the setting "Use UPnP / NAT-PMP to forward the port from my router" is always re-enabled.

## Steps to Reproduce

1. Create a qbittorrent container.
2. Connect to the WebUI.
3. Change the settings under WebUI for "Use UPnP / NAT-PMP to forward the port from my router", and/or try to enable HTTPS by providing a certificate and key.
4. Hit the save button in the container.
5. Restart the container, and watch how the settings revert...

## Environment
**OS:** Ubuntu LTS 20.04
**CPU architecture:** x86_64
**How docker service was installed:** Selected to run docker when installing Ubuntu.

## Command used to create docker container (run/create/compose/screenshot)

```
---
version: "2.1"
services:
  qbittorrent:
    image: lscr.io/linuxserver/qbittorrent
    container_name: qbittorrent
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Stockholm
      - WEBUI_PORT=8443
    volumes:
      - /home/pvz/config/qbittorrent:/config
      - /home/pvz/downloads:/downloads
      - /home/pvz/config/dehydrated/certs/server.example.com:/ssl:ro
    ports:
      - 8443:8443
    restart: unless-stopped
```

## Docker logs

```
qbittorrent    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
qbittorrent    | [s6-init] ensuring user provided files have correct perms...exited 0.
qbittorrent    | [fix-attrs.d] applying ownership & permissions fixes...
qbittorrent    | [fix-attrs.d] done.
qbittorrent    | [cont-init.d] executing container initialization scripts...
qbittorrent    | [cont-init.d] 01-envfile: executing...
qbittorrent    | [cont-init.d] 01-envfile: exited 0.
qbittorrent    | [cont-init.d] 01-migrations: executing...
qbittorrent    | [migrations] started
qbittorrent    | [migrations] no migrations found
qbittorrent    | [cont-init.d] 01-migrations: exited 0.
qbittorrent    | [cont-init.d] 10-adduser: executing...
qbittorrent    | usermod: no changes
qbittorrent    |
qbittorrent    | -------------------------------------
qbittorrent    |           _         ()
qbittorrent    |          | |  ___   _    __
qbittorrent    |          | | / __| | |  /  \
qbittorrent    |          | | \__ \ | | | () |
qbittorrent    |          |_| |___/ |_|  \__/
qbittorrent    |
qbittorrent    |
qbittorrent    | Brought to you by linuxserver.io
qbittorrent    | -------------------------------------
qbittorrent    |
qbittorrent    | To support LSIO projects visit:
qbittorrent    | https://www.linuxserver.io/donate/
qbittorrent    | -------------------------------------
qbittorrent    | GID/UID
qbittorrent    | -------------------------------------
qbittorrent    |
qbittorrent    | User uid:    1000
qbittorrent    | User gid:    1000
qbittorrent    | -------------------------------------
qbittorrent    |
qbittorrent    | [cont-init.d] 10-adduser: exited 0.
qbittorrent    | [cont-init.d] 30-config: executing...
qbittorrent    | [cont-init.d] 30-config: exited 0.
qbittorrent    | [cont-init.d] 90-custom-folders: executing...
qbittorrent    | [cont-init.d] 90-custom-folders: exited 0.
qbittorrent    | [cont-init.d] 99-custom-scripts: executing...
qbittorrent    | [custom-init] no custom files found exiting...
qbittorrent    | [cont-init.d] 99-custom-scripts: exited 0.
qbittorrent    | [cont-init.d] done.
qbittorrent    | [services.d] starting services
qbittorrent    | [services.d] done.
qbittorrent    | Catching signal: SIGTERM
qbittorrent    | Exiting cleanly
qbittorrent    |
qbittorrent    | ******** Information ********
qbittorrent    | To control qBittorrent, access the Web UI at http://localhost:8443
```

**Comment by j0nnymoe** [comment]
We don't recommend/support enabling SSL within the container. It's recommended to put it behind a reverse proxy to deal with SSL.

Regarding upnp, I'm surprised that it would work within a container and personally wouldn't have it enabled on your router anyways due to the usual security concerns.

**Comment by pv2b** [comment]
I have no idea how, but this has suddenly stopped happening to me.

It started with me standing up a new test container (with no shared volumes) to see if the problem occurred there, but it didn't. So then I tried reproducing it again in my "real" container, and suddenly the problem is no longer there.

I have no good explanation as to why, unfortunately. The only thing that I think COULD have caused this was I noticed a second qBittorrent.conf directly under /config (not under /config/qBittorrent) which I deleted, which I think I accidentally did when messing around with another container image and some misconfiguration.

Anyway, so the conclusion is that now suddenly I have this working. If there's a bug in here somewhere I've lost grip of it.
